src/handler/global/spawns.ts

```diff
@@ -0,0 +1,74 @@
+import { SlashCommandBuilder } from "@discordjs/builders";
+import { CommandInteraction, MessageEmbed } from "discord.js";
+import { LandAndFacilitiesRepository } from "../../repository/LandAndFacilitiesRepository";
+
+import { Command, CommandData } from "../../decorators";
+import { BotError } from "../../error";
+
+import { getCountries } from "../../map";
+
+import { FacQueryRow } from "../../repository/LandAndFacilitiesRepository";
+
+export class Spawns {
+    @CommandData({ type: "global" })
+    SpawnsData() {
+        return new SlashCommandBuilder()
+            .setName("spawns")
+            .setDescription("Check increase in all spawn's factory counts.")
+            .addIntegerOption((option) =>
+                option
+                    .setName("hours")
+                    .setDescription("How many hours change to check")
+                    .setRequired(true)
+            )
+            .toJSON();
+    }
+
+    @Command({ name: "spawns" })
+    async spawns(interaction: CommandInteraction): Promise<void> {
+        const hours = interaction.options.getInteger("hours") ?? 0;
+        if (hours <= 0) {
+            throw new BotError(
+                "Number of hours needs to be greater than zero!"
+            );
+        }
+        const hoursInMs = 60000 * 60 * hours;
+        const countryMap = await getCountries();
+        const facs = await LandAndFacilitiesRepository.getSpawnFactories(
+            new Date(Date.now() - hoursInMs)
+        );
+
+        const allies = facs.reduce<FacQueryRow[]>((acc, e) => {
```

Comment:
specifying the type here isn't necessary, I think

Reply:
actually, what are you even doing here. Why not use `filter`

Reply:
the type isnt necessary but vs code yells if its not there so i added it

Reply:
as to why i used reduce... uh i changed my approach and blindly reused code without thinking about it whoops will change lol